### PR TITLE
Status endpoint

### PR DESCRIPTION
**Pull request by jeffa5**
Add the status endpoint, initially useful for getting the version of the datastore.

### CODE REVIEW

File: proto/etcd.proto [hunk 1]
```diff
@@ -436,3 +436,35 @@ message MemberListResponse
   // members is a list of all members associated with the cluster.
   repeated Member members = 2;
 }
+
+message StatusRequest {}
+
+message StatusResponse
```
Comment (jumaffre) [inline proto/etcd.proto:442]: I think we need a note somewhere to say that this is mostly not implemented?

**Review by jumaffre — APPROVED** [review]

File: src/app/app.cpp [hunk 2]
```diff
@@ -1156,6 +1168,23 @@ namespace app
...
+      ccf::endpoints::ReadOnlyEndpointContext& ctx,
+      etcdserverpb::StatusRequest&& payload)
+    {
+      etcdserverpb::StatusResponse response;
+      CCF_APP_DEBUG("MAINTENANCE STATUS");
+
+      response.set_leader(0);
+      response.set_version(LSKV_VERSION);
+
+      // read a value from a map so our fill_header doesn't get skipped
```
[the remark this answers is not in the document]
Comment (jeffa5) [reply src/app/app.cpp:1181]: Difficult I think as the function is passed a `ccf::TxID` but one doesn't get created if we don't interact with the one given, maybe there should always be a txid instead? (Except for command endpoints I guess).

There might be a possibility to change this to a command endpoint anyway and handle the `fill_header` internally.

[the remark this answers is not in the document]
Comment (jumaffre) [reply src/app/app.cpp:1181]: A command endpoint would make sense here, but it's also not very urgent nor important at this point.

[the remark this answers is not in the document]
Comment (jeffa5) [reply src/app/app.cpp]: Updated it to be one, might as well get it out of the way now

**Pull request merged**